examples/dev-prod/src/main.rs

```diff
@@ -0,0 +1,61 @@
+use std::{
+    fmt::{self},
+    str::FromStr,
+};
+
+#[derive(PartialEq)]
+enum AppEnv {
+    Dev,
+    Prod,
+}
+
+/// A common setup that:
+///  - loads from a .env file in dev mode
+///  - loads from the envrironment in prod mode
+///
+/// A few commands to try:
+/// 1) `APP_ENV=prod HOST=prod.com cargo run`
+/// 2) `APP_ENV=dev HOST=prod.com cargo run`
+/// 3) `APP_ENV=prod cargo run`
```

Comment:
Maybe add just `cargo run` too.

Reply:
Example has been reordered:

1. `cargo run` - runs in dev mode, prints dev.com
2. `APP_ENV=prod cargo run` - errors because HOST is not set
3. `APP_ENV=prod HOST=prod.com cargo run` - prints prod.com